lib/dns.js

```diff
@@ -311,26 +318,31 @@ function setServers(servers) {
   const orig = cares.getServers();
   const newSet = [];
   const IPv6RE = /\[(.*)\]/;
-  const addrSplitRE = /:\d+$/;
+  const addrSplitRE = /(^.+?)(?::(\d+))?$/;
```

Comment:
> Not sure I follow? Just ignore any additional input. The host and the port are the only bits we care about. If you do not think it is appropriate to ignore the rest, then throw if values for the other parts are provided.

I think the issue is that this is our only validity check. Now `setServers(['8.8.8.8:53/abc?foo=bar'])` would throw. If we _only_ use `URL` it would be accepted with the `path` ignored (which is a future bug and `semver`ity issue)